deepchecks/vision/checks/distribution/train_test_label_drift.py

```diff
@@ -194,49 +182,129 @@ def condition(result: Dict) -> ConditionResult:
+        train_hists = calculate_continuous_histograms_in_batch(batch, train_hists, continuous_label_transformers,
+                                                               bounds, num_bins)
+        train_counters = calculate_discrete_histograms_in_batch(batch, train_counters, discrete_label_transformers)
+
+    for batch in test_dataset.get_data_loader():
+        test_hists = calculate_continuous_histograms_in_batch(batch, test_hists, continuous_label_transformers, bounds,
+                                                              num_bins)
+        test_counters = calculate_discrete_histograms_in_batch(batch, test_counters, discrete_label_transformers)
+
+    # Match discrete histograms to share x axis:
+    all_discrete_categories = [list(set(train_counter.keys()).union(set(test_counter.keys())))
+                               for train_counter, test_counter in zip(train_counters, test_counters)]
+
+    train_discrete_hists = iter([{k: train_counters[i][k] for k in all_discrete_categories[i]} for i in
+                                 range(num_discrete_transformers)])
+    test_discrete_hists = iter([{k: test_counters[i][k] for k in all_discrete_categories[i]} for i in
+                                range(num_discrete_transformers)])
+
+    # Transform continuous histograms into dict:
+    train_continuous_hists = iter([dict(zip(edges[i], train_hists[i])) for i in range(num_continuous_transformers)])
+    test_continuous_hists = iter([dict(zip(edges[i], test_hists[i])) for i in range(num_continuous_transformers)])
+
+    # # Return output in original order:
+    train_hists = [next(train_continuous_hists) if d['is_continuous'] is True else next(train_discrete_hists) for d in
+                   label_transformers]
+    test_hists = [next(test_continuous_hists) if d['is_continuous'] is True else next(test_discrete_hists) for d in
+                  label_transformers]
+
+    return train_hists, test_hists
+
+
+def calculate_discrete_histograms_in_batch(batch, counters, discrete_label_transformers):
+    """Calculate discrete histograms by batch."""
+    for i in range(len(discrete_label_transformers)):
+        calc_res = get_results_on_batch(batch, discrete_label_transformers[i])
+        counters[i].update(calc_res)
+    return counters
+
+
+def calculate_continuous_histograms_in_batch(batch, hists, continuous_label_transformers, bounds, num_bins):
+    """Calculate continuous histograms by batch."""
+    for i in range(len(continuous_label_transformers)):
+        calc_res = get_results_on_batch(batch, continuous_label_transformers[i])
+        new_hist, _ = np.histogram(calc_res, bins=num_bins, range=(bounds[i][0], bounds[i][1]))
+        hists[i] += new_hist
+    return hists
+
+
+def get_results_on_batch(batch, label_transformer):
+    """Calculate transformer result on batch of labels."""
+    list_of_arrays = batch[1]
+    calc_res = [label_transformer(arr) for arr in list_of_arrays]
```

Comment:
here comes the tricky part, you need to apply the label_transformer (the one from the dataset, that converts this to a common format) befor applying your label_transformer (which computes some quantity on it). That's why I think the latter should be renamed(:

Reply:
Done